Mf2/Parser.php

```diff
@@ -778,9 +835,14 @@ public function parseDT(\DOMElement $dt, &$dates = array()) {
 		 * if $dtValue is only a time and there are recently parsed dates,
 		 * form the full date-time using the most recently parsed dt- value
 		 */
-		if ((preg_match('/^\d{1,2}:\d{1,2}(Z?[+|-]\d{2}:?\d{2})?/', $dtValue) or preg_match('/^\d{1,2}[a|p]m/', $dtValue)) && !empty($dates)) {
+		if ((preg_match('/^\d{1,2}:\d{1,2}(Z?[+|-]\d{2}:?\d{2}?)?/', $dtValue) or preg_match('/^\d{1,2}[a|p]m/', $dtValue)) && !empty($dates)) {
```

Comment:
Hour-only timezone offsets (`+XX`, `-XX`) are valid according to the VCP spec. I updated the am/pm regex in both locations to handle optional periods as well `[ap]\.?m\.?`